languages/CHANGELOG_configure_help.en.txt
PHPGEDVIEW LANGUAGE FILE CHANGELOG

# $Id: CHANGELOG_configure_help.en.txt 7146 2016-12-06 20:06:25Z canajun2eh $

For information on changes prior to the release of PGV 4.2.0 (16 Nov. 2008), please
refer to the files in ZIP file "languages/CHANGELOG_bef.20081117.zip"

Use this file to follow the changes made to the English language files
between different versions of PhpGedView.  There are eight separate CHANGELOG files,
one to correspond to each file that together make up the entire set of English text.

The CHANGELOG files are:
		CHANGELOG_admin.en.txt
		CHANGELOG_configure_help.en.txt
		CHANGELOG_countries.en.txt
		CHANGELOG_editor.en.txt
		CHANGELOG_facts.en.txt
		CHANGELOG_faqlist.en.txt
		CHANGELOG_help_text.en.txt
		CHANGELOG_lang.en.txt

+$pgv_lang["keyword"] / means added
-$pgv_lang["keyword"] / deleted
*$pgv_lang["keyword"] / text was changed
#$pgv_lang["keyword"] / keyword was changed; Comment field will indicate the old keyword

Additional codes, to help Translators, follow the date as follows:
	A - Added or moved from another file
	C - Capitalization  (doesn't affect translation)
	D - Deleted
	f - Minor format change
	F - Major format change
	p - Punctuation change  (doesn't affect translation)
	M - Moved to another file
	R - keyword was renamed (no translation changes required)
	S - Spelling change		(doesn't affect translation)
	w - Minor wording change
	W - Major wording change or completely new text

The above list of codes may be amended as new situations arise.

VERSION 4.3.0

?$pgv_lang["whatever"]								dd Monthname, year		Change codes and comments
----------------------------------------------------------------------------------------
*$pgv_lang["USE_GEONAMES"]							06 Dec 2016				W
*$pgv_lang["USE_GEONAMES_help"]						06 Dec 2016				W
+$pgv_lang["GEOCODE_KEY"]							06 Dec 2016				A
+$pgv_lang["GEOCODE_KEY_help"]						06 Dec 2016				A
+$pgv_lang["GOOGLE_ANALYTICS_help"]		   			22 Aug 2011				A
+$pgv_lang["USE_GOOGLE_ANALYTICS_help"]	   			22 Aug 2011				A
+$pgv_lang["PGV_GOOGLE_ANALYTICS_help"]	   			22 Aug 2011				A
+$pgv_lang["PIWIK_ANALYTICS_help"]		   			22 Aug 2011				A
+$pgv_lang["USE_PIWIK_ANALYTICS_help"]	   			22 Aug 2011				A
+$pgv_lang["PGV_PIWIK_URL_help"]					22 Aug 2011				A
+$pgv_lang["PGV_PIWIK_SITE_help"]		   			22 Aug 2011				A
+$pgv_lang["CLUSTRMAPS_ANALYTICS_help"]	   			22 Aug 2011				A
+$pgv_lang["USE_CLUSTRMAPS_ANALYTICS_help"]			22 Aug 2011				A
+$pgv_lang["PGV_CLUSTRMAPS_SITE_help"]				22 Aug 2011				A
+$pgv_lang["PGV_CLUSTRMAPS_SERVER_help"]			22 Aug 2011				A

VERSION 4.2.4

?$pgv_lang["whatever"]								dd Monthname, year		Change codes and comments
----------------------------------------------------------------------------------------
+$pgv_lang["croatian"]								23 Mar 2010				A
*$pgv_lang["add_faq_pgvLangRefs"]					07 Feb 2010				W	Added example of a reference to a "PGV_..." constant
*$pgv_lang["um_bu_logs"]							24 Jan 2010				W
*$pgv_lang["USE_SILHOUETTE_help"]					05 Jan 2010				w added missing '>'
*$pgv_lang["USE_SILHOUETTE_help"]					03 Jan 2010				W
+$pgv_lang["USE_SILHOUETTE"]						02 Jan 2010				A
+$pgv_lang["USE_SILHOUETTE_help"]					02 Jan 2010				A


VERSION 4.2.3

?$pgv_lang["whatever"]								dd Monthname, year		Change codes and comments
-----------------------------------------------------------------------------------------
+$pgv_lang["lang_name_english (uk)"]				04 Nov 2009				A
+$pgv_lang["lang_name_english (us)"]				04 Nov 2009				A
-$pgv_lang["ALLOW_REMEMBER_ME_help"]				30 Sep 2009				D
-$pgv_lang["META_SURNAME_KEYWORDS"]					27 Sep 2009				D
-$pgv_lang["META_SURNAME_KEYWORDS_help"]			27 Sep 2009				D
*$pgv_lang["PGV_SMTP_HOST_help"]					10 Aug 2009				W
*$pgv_lang["PGV_SMTP_SSL_help"]						10 Aug 2009				W


VERSION 4.2.2

?$pgv_lang["whatever"]								dd Monthname, year		Change codes and comments
-----------------------------------------------------------------------------------------
*$pgv_lang["DB_UTF8_COLLATION_help"]				24 Jun 2009				w
-$pgv_lang["DBPERSIST"]								04 Jun 2009				D
-$pgv_lang["DBPERSIST_help"]			 			04 Jun 2009				D
*$pgv_lang["DBTYPE_help								16 May 2009				Sw
*$pgv_lang["DBTYPE_help								16 May 2009				W


VERSION 4.2.1

?$pgv_lang["whatever"]								dd Monthname, year		Change codes and comments
-----------------------------------------------------------------------------------------
*$pgv_lang["ENABLE_AUTOCOMPLETE"]					27 Mar 2009				A
*$pgv_lang["ENABLE_AUTOCOMPLETE_help"]				27 Mar 2009				A
*$pgv_lang["USE_THUMBS_MAIN_help"]					22 Mar 2009				W
*$pgv_lang["PGV_SMTP_SSL_help"]						15 Feb 2009				W
+$pgv_lang["PGV_SMTP_SSL_help"]						15 Feb 2009				A
*$pgv_lang["USE_THUMBS_MAIN_help"]					13 Feb 2009				w
+$pgv_lang["lang_name_serbian-la"]					12 Feb 2009				A


VERSION 4.2.0

?$pgv_lang["whatever"]								dd Monthname, year		Change codes and comments
-----------------------------------------------------------------------------------------
*$pgv_lang["PAGE_AFTER_LOGIN_help"]					30 Jan 2009				w
+$pgv_lang["PAGE_AFTER_LOGIN"]						29 Jan 2009				A
+$pgv_lang["PAGE_AFTER_LOGIN_help"]					29 Jan 2009				A
*$pgv_lang["PREFER_LEVEL2_SOURCES"]					26 Jan 2009				W
*$pgv_lang["USE_QUICK_UPDATE_help"]					26 Jan 2009				W
*$pgv_lang["USE_GEONAMES_help"]						25 Jan 2009				W
*$pgv_lang["USE_GEONAMES"]							24 Jan 2009				W
*$pgv_lang["USE_GEONAMES_help"]						24 Jan 2009				W
+$pgv_lang["USE_GEONAMES"]							24 Jan 2009				A
+$pgv_lang["USE_GEONAMES_help"]						24 Jan 2009				A
+$pgv_lang["lang_name_indonesian"]					24 Jan 2009				A
#$pgv_lang["more_config_hjaelp"]					21 Jan 2009				R	"more_config_hjaelp" --> "more_help_advice"
*$pgv_lang["general_privacy_help"]					21 Jan 2009				w	"more_config_hjaelp" --> "more_help_advice"
*$pgv_lang["PGV_SMTP_FROM_NAME_help"]				15 Jan 2009				W
+$pgv_lang["PGV_SMTP_FROM_NAME_help"]				15 Jan 2009				A
+$pgv_lang["lang_name_slovenian"]					7 Jan 2009				A
*$pgv_lang["PGV_SMTP_ACTIVE_help"]					22 Dec 2008				W
*$pgv_lang["PGV_SMTP_HOST_help"]					22 Dec 2008				W
*$pgv_lang["PGV_SMTP_HELO_help"]					22 Dec 2008				W
*$pgv_lang["PGV_SMTP_PORT_help"]					22 Dec 2008				W
*$pgv_lang["PGV_SMTP_AUTH_help"]					22 Dec 2008				W
*$pgv_lang["PGV_SMTP_AUTH_USER_help"]				22 Dec 2008				W
*$pgv_lang["PGV_SMTP_AUTH_PASS_help"]				22 Dec 2008				W
+$pgv_lang["PGV_SMTP_ACTIVE_help"]					22 Dec 2008				A
+$pgv_lang["PGV_SMTP_HOST_help"]					22 Dec 2008				A
+$pgv_lang["PGV_SMTP_HELO_help"]					22 Dec 2008				A
+$pgv_lang["PGV_SMTP_PORT_help"]					22 Dec 2008				A
+$pgv_lang["PGV_SMTP_AUTH_help"]					22 Dec 2008				A
+$pgv_lang["PGV_SMTP_AUTH_USER_help"]				22 Dec 2008				A
+$pgv_lang["PGV_SMTP_AUTH_PASS_help"]				22 Dec 2008				A
-$pgv_lang["USE_RTL_FUNCTIONS"]						21 Dec 2008				D
-$pgv_lang["USE_RTL_FUNCTIONS_help"]				21 Dec 2008				D
*$pgv_lang["person_privacy_help"]					30 Nov 2008				w	"ID 101" --> "ID I101"
*$pgv_lang["user_privacy_help"]						30 Nov 2008				w	"ID 100" --> "ID I100"
*$pgv_lang["edit_lang_utility_desc"]				28 Nov 2008				w	four --> nine
-$pgv_lang["media_import_help"]						28 Nov 2008				D
*$pgv_lang["compare_lang_utility_desc"]				28 Nov 2008				W	change wording about CHANGELOG files
*$pgv_lang["ENABLE_RSS_help"]						26 Nov 2008				w	Correct URL to Wiki article on RSS
*$pgv_lang["RSS_FORMAT_help"]						26 Nov 2008				w	Correct URL to Wiki article on RSS
*$pgv_lang["EDIT_AUTOCLOSE_help"]					24 Nov 2008				S
*$pgv_lang["DB_UTF8_COLLATION"]						24 Nov 2008				S	"UTF8" --> "UTF-8"
*$pgv_lang["DB_UTF8_COLLATION_help"]				24 Nov 2008				SW
*$pgv_lang["SURNAME_TRADITION_help"]				22 Nov 2008				W
*$pgv_lang["multi_letter_equiv_help"]				22 Nov 2008				CSf
*$pgv_lang["SURNAME_TRADITION_help"]				22 Nov 2008				W
+$pgv_lang["multi_letter_equiv"]					21 Nov 2008				A
+$pgv_lang["multi_letter_equiv_help"]				21 Nov 2008				A
+$pgv_lang["DB_UTF8_COLLATION"]						18 Nov 2008				A
+$pgv_lang["DB_UTF8_COLLATION_help"]				18 Nov 2008				A
+$pgv_lang["collation"]								17 Nov 2008				A
+$pgv_lang["collation_help"]						17 Nov 2008				A


changelog.txt
=====================================================================
	Change Log
	phpGedView

	This and other information can be found online at
	http://www.phpgedview.net

	# $Id: changelog.txt 7171 2017-11-05 16:18:39Z canajun2eh $
=====================================================================

Version 4.3.0
 - Fixed: Page footer links could end with an orphan ampersand or question-mark (canajun2eh)
 - Changed: Add more checks for hacking in incoming URI (canajun2eh)
 - Fixed: "Invalid GEDCOM format" error when saving an edited shared note (canajun2eh)
 - Changed: Speed improvement in reading 16-bit (256 colour) BMP images (canajun2eh)
 - Fixed: Undefined variable mod in module.php line 78 (canajun2eh)
 - Changed: Improve performance of BMP image support (canajun2eh)
 - Fixed: Syntax error in /includes/extras/functions.en.php (canajun2eh)
 - Changed: Move functions used only for debugging into their own functions_debug.php file (canajun2eh)
 - Added: Implement limited support for BMP images in media files -- can only write 24-bit images (canajun2eh)
 - Fixed: Correct generation of ordinal suffix in English (1st, 2nd, 3rd, etc.), French, and German (canajun2eh)
 - Fixed: "Download calendar file" not working in Todays Events, Upcoming Events, and Yahrzeiten blocks (canajun2eh)
 - Changed: Replaced /images/hcal.png with ical.png (canajun2eh)
 - Fixed: Yahrzeiten block does not obey "Allow download" configuration option (canajun2eh)
 - Fixed: ical.php not respecting language selection -- uses only English (canajun2eh)
 - Changed: Moved ical functions out of ical.php into their own file within includes/functions (canajun2eh)
 - Fixed: ical.php causes errors in class GedcomDate (canajun2eh)
 - Fixed: Person names should not contain comma when family name comes first (as in Hungarian) (canajun2eh)
 - Fixed: Undefined index "bulkMail" in includes/authentication.php when sending the Review Changes message (canajun2eh)
 - Changed: Make all language files HTML5 compliant by re-coding BIG, CENTER, FONT, and TT elements (canajun2eh)
 - Fixed: Help text for Customize Welcome page not displaying properly (canajun2eh)
 - Fixed: PHP7 warning about class constructor naming convention (canajun2eh)
 - Added: Check sending and receiving e-mail addresses for validity before trying to send (canajun2eh)
 - Updated: PHPMailer package updated to version 5.2.23 (canajun2eh)
 - Fixed: Mail send reported success even when sending actually failed (canajun2eh)
 - Fixed: Current SMTP port number does not display properly when reconfiguring (canajun2eh)
 - Fixed: Bulk messages to all users or inactive users not working (canajun2eh)
 - Fixed: PHP7 complaining about arithmetic involving strings passed to class_gedcomrecord by class_note (Canajun2eh)
 - Fixed: Undefined variable queryString in rss.php (canajun2eh)
 - Fixed: PHP7 complaining about arithmetic involving strings (Canajun2eh)
 - Fixed: PHP7 errors caused by incorrect usage of microtime() function (Philippe Conde)
 - Updated: More improvements to Googlemap module (Leonhard Zachl)
 - Fixed: Several errors in Googlemap module (tbabczynski)
 - Updated: Various improvements to Googlemap module (tbabczynski)
 - Fixed: Correct polish tradition of wife's surname in batch_update (tbabczynski)
 - Fixed: E-mails to admin sometimes sent in wrong language (canajun2eh)
 - Fixed: PHP 7 error when SALT is missing in call to crypt() function (canajun2eh)
 - Fixed: PHP 7 error in eval() function when it's fed the config.php file (canajun2eh)
 - Fixed: Mysql 5.7 error "1171 All parts of a PRIMARY KEY must be NOT NULL" (canajun2eh)
 - Fixed: Several instances of "array to string conversion" errors (canajun2eh)
 - Fixed: Call to undefined function cleanQueryString() in rss.php (canajun2eh)
 - Updated: Improved support for Googlemaps version 3 (Leonhard Zachl)
 - Fixed: User-to-user messaging ignores e-mail option (canajun2eh)
 - Fixed: Incorrect implementation of some messaging options in user profiles (canajun2eh)
 - Updated: Implement https method of calling Google charts and support for Googlemaps version 3 (Leonhard Zachl)
 - Fixed: German language files use double angle quotation marks incorrectly (canajun2eh)
 - Changed: Updates to PEAR.php and new PEAR5.php to accommodate more recent versions of PHP (Lester Caine)
 - Changed: Improve checks for improperly formed URI (canajun2eh)
 - Fixed: Incorrect call to static function in PEAR.php (Pasquale Ceres)
 - Changed: Remove "union" (a legitimate place name) from SQL injection check (canajun2eh)
 - Fixed: Adding a new media item while editing a Fact requires entry of the Indi, Family, or Source ID number (canajun2eh)
 - Fixed: Adding a new media item while editing a Fact causes "You are not welcome here" message (canajun2eh)
 - Changed: Automatically copy ID numbers of new shared notes, sources, and repositories (canajun2eh)
 - Fixed: Notes containing two at-symbols are sometimes incorrectly identified as referencing shared notes (canajun2eh)
 - Changed: Anonymous remote connections are now optional (Pasquale Ceres)
 - Fixed: SOAP errors cause out-of-memory errors (Pasquale Ceres)
 - Fixed: "update" should not be part of SQL injection check (canajun2eh)
 - Fixed: Import Progess bar should show percentage of total file size (canajun2eh)
 - Changed: Tighten SQL and cross-site script injection checks and relocate them to includes/session_spider.php (canajun2eh)
 - Fixed: Remote requests can return wrong results (Pasquale Ceres)
 - Fixed: Search returns wrong results when hits occur in non-genealogical tags (Pasquale Ceres)
 - Fixed: Wrong pseudo-icons for some file types, and Lightbox shouldn't try to handle HTM(L), DOC(X), PDF and TEX media files (canajun2eh)
 - Changed: General and Soundex searches, database checkboxes should be above search terms instead of below (canajun2eh)
 - Changed: Top menu "Search" should be using "post" method instead of "get", and field is too short (canajun2eh)

config.php
<?php
/**
 * Main configuration file required by all other files in PGV
 *
 * The variables in this file are the main configuration variable for the site
 * Gedcom specific configuration variables are stored in the config_gedcom.php file.
 * Site administrators may edit these settings online through the install.php file.
 *
 * phpGedView: Genealogy Viewer
 * Copyright (C) 2002 to 2009  PGV Development Team.  All rights reserved.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *
 * @package PhpGedView
 * @subpackage Admin
 * @see install.php
 * @version $Id: config.dist 7054 2011-08-22 19:30:30Z canajun2eh $
 */

if (preg_match('/\Wconfig.php/', $_SERVER['SCRIPT_NAME'])>0) {
	print 'Got your hand caught in the cookie jar.';
	exit;
}

$DBTYPE = "mysql";								//-- type of database to connect when using the PHP/PDO
$DBHOST = "localhost";							//-- Host where MySQL database is kept
$DBPORT = "";									//-- Database port, leave blank for default
$DBUSER = "root";								//-- MySQL database User Name
$DBPASS = "";									//-- MySQL database User Password
$DBNAME = "phpgedview";							//-- The MySQL database name where you want PHPGedView to build its tables
$DB_UTF8_COLLATION = false;						//-- Use the database to sort/collation UTF8 text
$TBLPREFIX = "pgv_";							//-- prefix to include on table names
$INDEX_DIRECTORY = "./index/";					//-- Readable and Writeable Directory to store index files (include the trailing "/")
$AUTHENTICATION_MODULE = "authentication.php";	//-- File that contains authentication functions
$PGV_STORE_MESSAGES = true;						//-- allow messages sent to users to be stored in the PGV system
$PGV_SIMPLE_MAIL = true;						//-- allow admins to set this so that they can override the name <emailaddress> combination in the emails
$USE_REGISTRATION_MODULE = true;				//-- turn on the user self registration module
$REQUIRE_ADMIN_AUTH_REGISTRATION = true;		//-- require an admin user to authorize a new registration before a user can login
$ALLOW_USER_THEMES = true;						//-- Allow user to set their own theme
$ALLOW_CHANGE_GEDCOM = true;					//-- A true value will provide a link in the footer to allow users to change the gedcom they are viewing
$LOGFILE_CREATE = "monthly";					//-- set how often new log files are created, "none" turns logs off, "daily", "weekly", "monthly", "yearly"
$LOG_LANG_ERROR = false;						//-- Set if non-existing language variables should be written to a logfile
$PGV_SESSION_SAVE_PATH = "";					//-- Path to save PHP session Files -- DO NOT MODIFY unless you know what you are doing
												//-- leaving it blank will use the default path for your php configuration as found in php.ini
$PGV_SESSION_TIME = "7200";						//-- number of seconds to wait before an inactive session times out
$SERVER_URL = "";								//-- the URL used to access this server
$LOGIN_URL = "";								//-- the URL to use to go to the login page, use this value if you want to redirect to a different site when users login, useful for switching from http to https
$MAX_VIEWS = "20";								//-- the maximum number of page views per xx seconds per session
$MAX_VIEW_TIME = "1";							//-- the number of seconds in which the maximum number of views must not be reached
$PGV_MEMORY_LIMIT = "32M";						//-- the maximum amount of memory that PGV should be allowed to consume
$COMMIT_COMMAND = "";							//-- Choices are empty string, cvs or svn
$PGV_SMTP_ACTIVE = false;						//-- Enable to use of SMTP to send external emails
$PGV_SMTP_HOST = "";							//-- The domain name of the SMTP smarthost
$PGV_SMTP_HELO = "";							//-- The local SMTP domain name
$PGV_SMTP_PORT = "25";							//-- The port number talk to the SMTP smarthost
$PGV_SMTP_AUTH = false;							//-- Enable the use of SMTP authorization
$PGV_SMTP_AUTH_USER = "";						//-- User name for SMTP authorization
$PGV_SMTP_AUTH_PASS = "";						//-- Password for SMTP authorization
$PGV_SMTP_SSL = false;							//-- Use SSL for SMTP authorization
$PGV_SMTP_FROM_NAME = "";						//-- Sender name
$USE_GOOGLE_ANALYTICS = false;					//-- Should this site link to Google Analytics server?
$PGV_GOOGLE_ANALYTICS = "";						//-- This PGV site's Google Analytics account
$USE_PIWIK_ANALYTICS = false;					//-- Should this site link to Piwik Analytics server?
$PGV_PIWIK_URL = "";							//-- URL to Piwik Analytics server
$PGV_PIWIK_SITE = "";							//-- This PGV site's site number on Piwik Analytics server
$USE_CLUSTRMAPS_ANALYTICS = false;				//-- Should this site link to ClustrMaps Analytics server?
$PGV_CLUSTRMAPS_SITE = "";						//-- This PGV site's URL, for use by ClustrMaps Analytics server
$PGV_CLUSTRMAPS_SERVER = "";					//-- ClustrMaps server number assigned to handle your site's requests

$CONFIGURED = false;
require_once './includes/session.php';
